asciidoctorj-cli/src/main/java/org/asciidoctor/cli/AsciidoctorCliOptions.java

```diff
@@ -266,88 +276,77 @@ private boolean isInPlaceRequired() {
+        if (this.safeMode != null) {
+            opts.put(ruby.newSymbol("safe"), this.safeMode.getLevel());
         }
-
         if (this.noHeaderFooter) {
-            optionsBuilder.option(Options.STANDALONE, false);
+            opts.put(ruby.newSymbol(Options.STANDALONE), false);
         }
-
-        if (this.compact) {
-            optionsBuilder.compact(this.compact);
+        if (this.sectionNumbers) {
+            attributes.put("sectnums", "");
         }
-
-        if (isBaseDirOption()) {
-            optionsBuilder.baseDir(new File(this.baseDir).getCanonicalFile());
+        if (this.eruby != null) {
+            opts.put(ruby.newSymbol(Options.ERUBY), this.eruby);
+        }
+        if (isTemplateDirOption()) {
+            opts.put(ruby.newSymbol(Options.TEMPLATE_DIRS), this.templateDir.toArray(new String[0]));
         }
-
         if (isTemplateEngineOption()) {
-            optionsBuilder.templateEngine(this.templateEngine);
+            opts.put(ruby.newSymbol(Options.TEMPLATE_ENGINE), this.templateEngine);
         }
-
-        if (isTemplateDirOption()) {
-            for (String templateDir : this.templateDir) {
-                optionsBuilder.templateDirs(new File(templateDir).getCanonicalFile());
-            }
+        if (this.baseDir != null) {
+            opts.put(ruby.newSymbol(Options.BASEDIR), this.baseDir);
         }
-
-        if (isDestinationDirOption() && !isOutputStdout()) {
-            optionsBuilder.toDir(new File(this.destinationDir).getCanonicalFile());
-
-            if (isSourceDirOption()) {
-                optionsBuilder.sourceDir(new File(this.sourceDir).getCanonicalFile());
-            }
+        if (this.destinationDir != null) {
+            opts.put(ruby.newSymbol("destination_dir"), this.destinationDir);
         }
-
-        if (isInPlaceRequired()) {
-            optionsBuilder.inPlace(true);
+        if (this.trace) {
+            opts.put(ruby.newSymbol(Options.TRACE), true);
         }
-
-        Attributes attributesBuilder = buildAttributes();
-        if (this.sectionNumbers) {
-            attributesBuilder.setSectionNumbers(this.sectionNumbers);
+        if (this.timings) {
+            opts.put(ruby.newSymbol("timings"), true);
+        }
+        if (this.warnings) {
+            opts.put(ruby.newSymbol(Options.WARNINGS), true);
+        }
+        if (!attributes.isEmpty()) {
+            opts.put(ruby.newSymbol(Options.ATTRIBUTES), attributes);
+        }
+        if (this.isSourceDirOption()) {
+            opts.put(ruby.newSymbol("source_dir"), this.sourceDir);
         }
-        optionsBuilder.attributes(attributesBuilder);
-        return optionsBuilder.build();
+        return opts;
     }
 
     /**
      * Returns the attributes injected as List<String> into a structured
      * {@link Attributes} instance.
      */
     // FIXME Should be private, made protected for testing.
-    Attributes buildAttributes() {
+    Map buildAttributes() {
```

Comment:
The CLI would not be anything that we would expose publicly, no?
Since we don't invoke Asciidoctor via the AsciidoctorJ API anymore with this PR, we would need to pass a map (Hash) to Asciidoctor via the option `attributes`.

Reply:
It's more a "dogfooding" thing, but if there's no better way, it's fine.

Reply:
Changed back to using Attributes.

Reply:
Thanks, I hope this makes sense and you are not doing this to appease me.

Reply:
:D No worries.
I think at some point we need to do some changes to our implementations of Options and Attributes.
Right now we have `map()` deprecated but still have to use it ourselves.